callstats-java-sdk/src/main/java/io/callstats/sdk/CallStatsHttpClient.java

```diff
@@ -0,0 +1,233 @@
+	}
+	
+	
+	public String getBaseUrl() {
+		return BASE_URL;
+	}
+	
+	public CallStatsHttpClient() {
+		super();  
+		Properties prop = new Properties();
+		InputStream input = null;
+	     
+    
+		input = getClass().getClassLoader().getResourceAsStream(CallStatsConst.CallStatsJavaSDKPropertyFileName);
+		if(input !=null){
+			try {
+				prop.load(input);
+				BASE_URL = prop.getProperty("CallStats.BaseURL");
+				connectionTimeOut = Integer.parseInt(prop.getProperty("CallStats.ConnectionTimeOut"));
+				soTimeOut = Integer.parseInt(prop.getProperty("CallStats.SOTimeOut"));
+			} catch (IOException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}
+		}
+
+		
+		logger.info("Base URL is "+BASE_URL);
+        // Create I/O reactor configuration
+        IOReactorConfig ioReactorConfig = IOReactorConfig.custom()
+                .setIoThreadCount(Runtime.getRuntime().availableProcessors())
+                .setConnectTimeout(connectionTimeOut)
+                .setSoTimeout(soTimeOut)
+                .build();
+
+        // Create a custom I/O reactort
+        ConnectingIOReactor ioReactor;
+		try {
+			ioReactor = new DefaultConnectingIOReactor(ioReactorConfig);
+		} catch (IOReactorException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+			throw new RuntimeException(e);
+		}
+        
+        
+		PoolingHttpClientConnectionManager mgr;
+		mgr = new PoolingHttpClientConnectionManager();
+		((PoolingHttpClientConnectionManager) mgr).setDefaultMaxPerRoute(10);
+		httpclient = HttpClients.custom()
+	            .setConnectionManager(mgr)
+	            .build();
+		
+		PoolingNHttpClientConnectionManager asyncMgr;
+		asyncMgr = new PoolingNHttpClientConnectionManager(ioReactor);
+		httpAsyncClient = HttpAsyncClients.custom()
+				.setConnectionManager(asyncMgr)
+	            .build();
+	}
+	
+	private HttpPost generateHttpPostRequest(String url, String body) {
+		URI uri = generateURI(url);
+		StringEntity entity = generateEntity(body);
+
+		HttpPost post = new HttpPost(uri);
+		post.setHeader("Content-type", "application/json");
+	    post.setHeader("Accept", "application/json");
+		post.setEntity(entity);
+
+		return post;
+	}
+	
+	private URI generateURI(String url) {
+		URI uri;
+		try {
+			uri = new URI(url);
+		} catch (URISyntaxException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+			throw new IllegalStateException("Invalid URL:"+url, e);
+		}
+		return uri;
+	}
+	
+	private StringEntity generateEntity(String body) {
+		StringEntity entity;
+		try {
+			entity = new StringEntity(body);
+		} catch (UnsupportedEncodingException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+			throw new RuntimeException(e);
+		}
+		entity.setContentType("application/json");		
+		return entity;
+	}
+		
+	public HttpResponse sendHttpRequest(String url, String httpMethodType, String body) {
+		
+		if(url == null || httpMethodType == null || body == null) {
```

Comment:
I'm sure there're java libraries to check a String instance is either null or empty. So better use those library here.